### PR TITLE
feat(rl-components): add jaxatari environment substrate

### LINKED ISSUE #10: Baseline: Integrate JAXAtari and DeepMind Preprocessing
Integrate `JAXAtari` (https://github.com/k4ntz/JAXAtari) into the workspace. Create a JAX-native environment wrapper in `rl-components` that applies the standard DeepMind preprocessing pipeline: Frame Stacking (4 frames), Grayscale conversion, MaxPool over skipped frames (action repeat 4), and life loss as terminal state.

### PR DESCRIPTION
**Pull request by andnp**
## Summary
- add a canonical environment protocol in `rl-components`
- add a JAXAtari adapter with DeepMind-style preprocessing defaults
- add real JAXAtari smoke coverage in CI with scripted asset installation
- add a Gymnax compatibility bridge for incremental consumer adoption
- add benchmark-based env-seam regression detection in CI

## Details
- introduced `EnvSpec`, `EnvReset`, `EnvStep`, and `EnvProtocol`
- pinned `jaxatari` to GitHub tag `v0.1`
- added `JAXAtariAdapter` and `JAXAtariConfig` in `rl_components.atari`
- added a real smoke test plus `scripts/install_jaxatari_assets.py`
- added `GymnaxCompatibilityBridge` to let current Gymnax-style consumers adopt canonical envs incrementally
- added a dedicated benchmark job using `pytest-benchmark` and `benchmark-action/github-action-benchmark`

## Verification
- `uv run pytest tests/small/test_rl_components_env_protocol.py -q`
- `uv run pytest tests/small/test_rl_components_atari.py tests/medium/test_rl_components_atari_jit.py -q`
- `uv run pytest tests/small/test_rl_components_gymnax_bridge.py tests/medium/test_rl_components_gymnax_bridge_jit.py -q`
- `uv run pytest tests/regression/test_rl_components_atari_real_smoke.py -q`
- `JAXATARI_CONFIRM_OWNERSHIP=1 XDG_DATA_HOME=<temp> uv run python scripts/install_jaxatari_assets.py`
- `JAXATARI_RUN_SMOKE=1 XDG_DATA_HOME=<same temp> uv run pytest -q tests/regression/test_rl_components_atari_real_smoke.py::test_real_jaxatari_adapter_smoke -x`
- `uv run pytest --benchmark-only --benchmark-json output.json tests/performance/test_rl_components_gymnax_bridge_bench.py -q`
- `uv run ruff check libs/rl-components/src/rl_components/env_protocol.py libs/rl-components/src/rl_components/atari.py libs/rl-components/src/rl_components/gymnax_bridge.py tests/small/test_rl_components_env_protocol.py tests/small/test_rl_components_atari.py tests/medium/test_rl_components_atari_jit.py tests/small/test_rl_components_gymnax_bridge.py tests/medium/test_rl_components_gymnax_bridge_jit.py tests/regression/test_rl_components_atari_real_smoke.py tests/performance/test_rl_components_gymnax_bridge_bench.py scripts/install_jaxatari_assets.py`
- `uv run pyright libs/rl-components/src/rl_components/env_protocol.py libs/rl-components/src/rl_components/atari.py libs/rl-components/src/rl_components/gymnax_bridge.py tests/small/test_rl_components_env_protocol.py tests/small/test_rl_components_atari.py tests/medium/test_rl_components_atari_jit.py tests/small/test_rl_components_gymnax_bridge.py tests/medium/test_rl_components_gymnax_bridge_jit.py tests/regression/test_rl_components_atari_real_smoke.py tests/performance/test_rl_components_gymnax_bridge_bench.py scripts/install_jaxatari_assets.py`
- `uv run ty check libs/rl-components/src/rl_components/env_protocol.py libs/rl-components/src/rl_components/atari.py libs/rl-components/src/rl_components/gymnax_bridge.py tests/small/test_rl_components_atari.py tests/medium/test_rl_components_atari_jit.py tests/small/test_rl_components_gymnax_bridge.py tests/medium/test_rl_components_gymnax_bridge_jit.py tests/performance/test_rl_components_gymnax_bridge_bench.py`

Closes #10

### CODE REVIEW

File: docs/specs/canonical-env-protocol.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
+# Technical Specification: Canonical Environment Protocol
+
+## 1. Overview
+`rl-components` needs a small internal environment protocol that keeps shared RL code backend-agnostic while remaining fully compatible with end-to-end JAX execution. The canonical abstraction is a single-environment interface with explicit `reset`, `step`, and `spec` entry points.
+
+This protocol is the substrate for issue #10. It does not add adapters yet. It only defines the shared contract that later adapters can implement.
```
Comment (andnp) [inline docs/specs/canonical-env-protocol.md]: Let's avoid "temporal" comments like that. They never age well. There are a few in this document that we should remove.

File: libs/rl-components/src/rl_components/env_protocol.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        def __init__(
+            self,
+            *,
+            id: str,
+            observation_shape: tuple[int, ...],
+            action_shape: tuple[int, ...],
+            observation_dtype: jnp.dtype = jnp.float32,
+            action_dtype: jnp.dtype = jnp.int32,
+            num_actions: int | None = None,
+        ) -> None: ...
```
Comment (andnp) [inline libs/rl-components/src/rl_components/env_protocol.py]: This is quite unfortunate. Since we're going to use `chex` a lot, this definitely suggests that we need to find a cleaner and more scalable way out of this pattern. 

Is there value in wrapping `chex.dataclass` in our own utility library and improving the type inference ourselves?
